packages/pn-validator/src/types/TypeRules.ts

```diff
@@ -1,43 +1,20 @@
+import { BooleanRuleValidator } from './../ruleValidators/BooleanRuleValidator';
 import { NumberRuleValidator } from './../ruleValidators/NumberRuleValidator';
 import { StringRuleValidator } from './../ruleValidators/StringRuleValidator';
 import { DateRuleValidator } from './../ruleValidators/DateRuleValidator';
 import { ObjectRuleValidator } from '../ruleValidators/ObjectRuleValidator';
 import { ArrayRuleValidator } from './../ruleValidators/ArrayRuleValidator';
-import { CommonRuleValidator } from '../ruleValidators/CommonRuleValidator';
 
-type MixedType = String | Number | Date | Boolean;
-
-type StringTypeRules<TModel, TValue> = [TValue] extends [String | undefined | null]
-  ? StringRuleValidator<TModel, TValue>
-  : {};
-
-type NumberTypeRules<TModel, TValue> = [TValue] extends [Number | undefined | null]
-  ? NumberRuleValidator<TModel, TValue>
-  : {};
-
-type DateTypeRules<TModel, TValue> = [TValue] extends [Date | undefined | null]
-  ? DateRuleValidator<TModel, TValue>
-  : {};
-
-type BoolenaTypeRules<TModel, TValue> = [TValue] extends [Boolean | undefined | null]
-  ? CommonRuleValidator<TModel, TValue>
-  : {};
-
-type ArrayTypeRules<TModel, TValue> = [TValue] extends [Array<infer _TEachValue> | undefined | null]
-  ? ArrayRuleValidator<TModel, TValue>
+export type TypeRules<TModel, TValue> = [TValue] extends [String | undefined | null]
```

Comment:
I remember that we tried several variants of this definition. Maybe a comment about what it should be taken with care about any possible changes (i.e. keep the `| undefined | null` in each type, and just one `{}` at the end) would help other people understand the form of this definition.

Reply:
Done